include/ddc/kernels/splines/spline_builder.hpp

```diff
@@ -599,10 +600,36 @@ void SplineBuilder<
         return;
 	*/
 
-    matrix = ddc::detail::SplinesLinearProblemMaker::make_new_sparse<ExecSpace>(
-            ddc::discrete_space<BSplines>().nbasis(),
-            cols_per_chunk,
-            preconditionner_max_block_size);
+    if constexpr (Solver == ddc::SplineSolver::LAPACK) {
+        int upper_band_width;
+        if (bsplines_type::is_uniform()) {
+            upper_band_width = bsplines_type::degree() / 2;
+        } else {
+            upper_band_width = bsplines_type::degree() - 1;
+        }
+        if constexpr (bsplines_type::is_periodic()) {
+            matrix = ddc::detail::SplinesLinearProblemMaker::make_new_periodic_band_matrix<
+                    ExecSpace>(
+                    ddc::discrete_space<BSplines>().nbasis(),
+                    upper_band_width,
+                    upper_band_width,
+                    bsplines_type::is_uniform());
+        } else {
+            matrix = ddc::detail::SplinesLinearProblemMaker::
+                    make_new_block_matrix_with_band_main_block<ExecSpace>(
+                            ddc::discrete_space<BSplines>().nbasis(),
+                            upper_band_width,
+                            upper_band_width,
+                            bsplines_type::is_uniform(),
+                            lower_block_size,
+                            upper_block_size);
```

Comment:
```suggestion
                            lower_block_size,
                            upper_block_size);
```

Was there a bug in sll https://gitlab.maisondelasimulation.fr/gysela-developpers/gyselalibxx/-/blob/2fe50d20ff196a4d8a21cfc9a1c35021aa78cbd8/vendor/sll/include/sll/spline_builder.hpp#L340 ?

Reply:
No, both are possible but I think I changed the convention for bad reasons at some point (upper should be before lower in an arguments list because we describe matrices from the top to the bottom)

Reply:
Actually both are consistent. The reason I had to put `lower_block_size `first here is because I want `top_left_size` to have a default value there:

https://github.com/CExA-project/ddc/blob/729cd846a8c022eba605c281f1cb28c7ed8ade24/include/ddc/kernels/splines/splines_linear_problem_maker.hpp#L90

Reply:
just that default value has to be last in the argument list